src/Template/Element/sentences/audio.ctp

```diff
@@ -25,64 +25,79 @@
+$audiosJson = json_encode($audios);
+
+// Prevent interpolation by AngularJS
+$audioLicenses = str_replace('{{', '\{\{', $audioLicenses);
+$audiosJson = str_replace('{{', '\{\{', $audiosJson);
+
+$this->Html->script('/js/sentences/audio-details.ctrl.js', ['block' => 'scriptBottom']);
+$this->Html->script('/js/directives/audio-button.dir.js', ['block' => 'scriptBottom']);
+$this->AngularTemplate->addTemplate(
+    $this->element('sentence_buttons/audio'),
+    'audio-button-template'
+);
 ?>
-<div class="section md-whiteframe-1dp">
+<div ng-controller="AudioDetailsController as vm"
+     ng-init="vm.init(<?= h($audiosJson) ?>, <?= h($audioLicenses) ?>)"
+     ng-cloak>
+<div ng-if="vm.audios.length > 0"
+     layout="column"
+     class="section audio md-whiteframe-1dp">
     <?php /* @translators: header text in sentence page */ ?>
-    <h2><?php echo __('Audio') ?></h2>
-<?php
+    <h2><?= __n('Audio', 'Audio', count($audios)) ?></h2>
 
-if ($hasaudio) {
-    $this->Audio->displayAudioInfo($audios[0]);
-}
+    <div ng-repeat="audio in vm.audios" ng-class="{'disabled': !audio.enabled}">
+        <h3>
+            <audio-button include-disabled="true" audios="[audio]"></audio-button>
+            <span>
+                <?= format(__('by {username}'), [
+                    'username' => '<a ng-href="{{audio.attribution_url}}">{{audio.author}}</a>'
+                ]) ?>
+            </span>
+        </h3>
 
-if (CurrentUser::isAdmin()) {
-    if ($hasaudio) {
-        echo "<hr>";
-    }
-    echo $this->Form->create(
-        "Sentence",
-        array(
-            "url" => array("action" => "edit_audio"),
-            "type" => "post",
-        )
-    );
-    echo $this->Form->hidden(
-        "id",
-        array("value" => $sentenceId)
-    );
-    echo __d("admin", "Enabled");
-    echo $this->Form->control(
-        "hasaudio",
-        array(
-            "label" => false,
-            "type" => "radio",
-            "options" => array(
-                1 => __d("admin", "Yes"),
-                0 => __d("admin", "No")
-            ),
-            "value" => $hasaudio
-        )
-    );
+        <div class="audio-details">
+            <div class="license"><?= format(__('License: {license}'), ['license' => $licenseTemplate]) ?></div>
 
-    $ownerName = '';
-    if ($hasaudio) {
-        $ownerName = $this->safeForAngular($audios[0]->author);
-    }
-    echo $this->Form->control("ownerName",
-        array(
-            'label' => __d("admin", "Owner name"),
-            "value" => $ownerName
-        )
-    );
-    echo $this->Form->submit(__d('admin', 'Submit'));
-    echo $this->Form->end();
-}
-?>
+            <?php if (CurrentUser::isAdmin()): ?>
+                <md-checkbox
+                    ng-model="audio.enabled"
+                    class="md-primary">
+                    <?= __('Is enabled') ?>
+                </md-checkbox>
+                <md-input-container class="md-button-right">
```

Comment:
What is the `md-button-right` here for?

Also, not sure where else to mention this, but after saving the form, the page reloads in a partially scrolled state with the audio card not yet filled in, which means that the logs show up slightly higher on the page until AngularJS finishes rendering and then after clicking this input box, the scroll position jumps to where the logs are. (At least in my Firefox on Desktop, but not Android.) Not sure whether that's an issue with AngularJS trying and failing to scroll the focused input into view, or whether it's a problem with Firefox.

Reply:
`md-button-right` was an unfortunate copy-paste :grinning:

Reply:
About the scroll jump, I was able to reproduce the bug on Firefox desktop, too. But not on Chromium. The impact of this bug is so minor that I won’t bother to investigate further.

Reply:
Fair.